services/collaboration/README.md

```diff
@@ -0,0 +1,343 @@
+  ensuring active collaboration sessions are properly managed.
+- `WebSocket-based Communication`: Uses WebSocket connections to handle real-time synchronization of Yjs documents
+  between users.
+- `MongoDB Persistence`: Yjs document updates and room data are persisted in MongoDB, ensuring fault tolerance and the
+  ability to resume sessions after interruptions.
+- `Automatic Cleanup`: When a room is closed, the corresponding Yjs document is removed from MongoDB, ensuring efficient
+  use of resources.
+
+---
+
+## Environment Variables
+
+Here are the key environment variables used in the `.env` file:
+
+| Variable                 | Description                                                                           |
+|--------------------------|---------------------------------------------------------------------------------------|
+| `COLLAB_CLOUD_MONGO_URI` | URI for connecting to the MongoDB Atlas database for the collaboration service (room) |
+| `COLLAB_LOCAL_MONGO_URI` | URI for connecting to the local MongoDB database for the collaboration service (room) |
+| `YJS_CLOUD_MONGO_URI`    | URI for connecting to the MongoDB Atlas database for Yjs document persistence         |
+| `YJS_LOCAL_MONGO_URI`    | URI for connecting to the local MongoDB database for Yjs document persistence         |
+| `DB_USERNAME`            | Username for the MongoDB databases (for both cloud and local environments)            |
+| `DB_PASSWORD`            | Password for the MongoDB databases (for both cloud and local environments)            |
+| `QUESTION_SERVICE_URL`   | URL for connecting to the Question Service API                                        |
+| `CORS_ORIGIN`            | Allowed origins for CORS (default: * to allow all origins)                            |
+| `COLLAB_WS_PORT`         | Port for the WebSocket server (default: 8084)                                         |
+| `COLLAB_HTTP_PORT`       | Port for the HTTP server (default: 8087)                                              |
+| `WEBSOCKET_AUTH_FAILED`  | WebSocket error code for failed authorization (default: 4000)                         |
+| `WEBSOCKET_ROOM_CLOSED`  | WebSocket error code for closed rooms (default: 4001)                                 |
+| `ENV`                    | Environment setting (`development` or `production`)                                   |
+
+---
+
+## Documentation on API Endpoints
+
+The `Collaboration Service` provides HTTP API endpoints to manage and retrieve details about rooms used in the real-time
+collaboration service. It enables creating rooms, retrieving room details, and managing room statuses.
+
+---
+
+## Get Room IDs by User ID
+
+This endpoint retrieves all room IDs for a given user, but only if the room is still active (`room_status` is `true`).
+One user can have multiple rooms, and each room is identified by a unique `room_id`.
+
+- **HTTP Method**: `GET`
+- **Endpoint**: `/room/user/{userId}`
```

Comment:
For all methods by `userId`, we should not allow them to specify the `userId` themselves. This means that they can access the endpoints for any user. Instead, we should make use of the authorization token that is provided with each request, and using the `JWT_TOKEN`, extract the `userId`.